core/src/regex.rs

```diff
@@ -90,6 +94,10 @@ fn rust_array_decl_struct() -> Regex {
 	Regex::new(r"^\[ *?([\w><]+) *?; *?(\d+) *?\]").expect("Primitive Regex expression invalid")
 }
 
+pub fn rust_bit_size() -> Regex {
+	Regex::new(r"^(Int|UInt)<([\w\d]+), ?([\w\d]+)>").expect("Regex expression should be infallible; qed")
```

Comment:
I guess this regex is for `Int<size, type>`, I'm not a regex export but...

I'm surprised that `[\d\w]+` is used for the size, it would match `beef999bibmap` for example. Thus, if `size` is in decimal just use `\d` then.

Why `, ?([\w\d]+)`  it would only work for `Int<999,bar>` and `Int<999, bar` not for arbitrary number of spaces :)
I suggest `, *(\w+)` instead.

Reply:
I don't think the capture groups are necessary for the type params, i.e. this should be fine: `(Int|UInt)<[\w\d]+, *[\w\d]+>` and like niklas I'm not sure I understand what the input to this looks like: can both params contain both letters and number? A test that exercise this case would be good.

Reply:
Here's a sample of some of the input from polkadot.js:
```JSON     
      "Fixed64": "Int<64, Fixed64>",
      "FixedI64": "Int<64, FixedI64>",
      "FixedU64": "UInt<64, FixedU64>",
      "Fixed128": "Int<128, Fixed128>",
      "FixedI128": "Int<128, FixedI128>",
      "FixedU128": "UInt<128, FixedU128>",
      "I32F32": "Int<64, I32F32>",
      "U32F32": "UInt<64, U32F32>",
      "PerU16": "UInt<16, PerU16>",
      "Perbill": "UInt<32, Perbill>",
      "Percent": "UInt<8, Percent>",
      "Permill": "UInt<32, Permill>",
      "Perquintill": "UInt<64, Perquintill>",
      "Balance": "UInt<128, Balance>",
```


So the second type parameter is unnecessary, since it's already a key in our de-serialized JSON, but the first type parameter is needed to delegate the size of the type we are dealing with. This also leads me to believe that the first type parameter will always be a digit and the second just a repeat of the JSON key (I haven't found anything in the definitions that contradicts this). I'm not 100% on what the rationale of this change is in Polkadot.JS, but in the PR's from Polkadot.js I found (like [here](https://github.com/polkadot-js/api/pull/1891/files) they use the Type<BitLength> in that style.

I agree with @niklasad1 , though, arbitrary number of spaces is definitely preferable! fixing that

Reply:
All right, I see

We could do `(Int|UInt)<[\d]+(, *[\w\d]+)?> instead to make the second type parameter optional then I guess